Sources/App/Commands/Analyze.swift

```diff
@@ -239,8 +239,20 @@ extension Analyze {
 
             let versions = try await updateLatestVersions(on: tx, package: package)
 
-            updateScore(package: package, versions: versions)
+            let targets: [(String, TargetType)]? = try await {
+                guard let repo = package.repository,
+                      let owner = repo.owner,
+                      let repository = repo.name
+                else { return nil }
+
+//                let thisLineCausesProcessingTestFailure = try await API.PackageController.Target
+//                    .query(on: database, owner: owner, repository: repository)
```

Comment:
So, I pushed a debugging commit in 4dec651f8735fd729895305ff8a7bf84f770ccfe because I've narrowed the test failure down to this line of code. What we actually want to do here is return these target name/type pairs, but just executing that query causes `test_processing_pipeline` to fail.

With the commented-out line included, the test fails (even still returning the `[]`). When it’s commented out, the test passes.

I’ve traced that query back to its root, and it’s just a `SELECT` query. I can’t imagine how it’d affect the processing test in the way it is, except maybe a timing issue from adding the extra time for that query?

Any ideas, @finestructure?

Reply:
The problem is that this query is throwing (with `connectionRequestTimeout` for me, that's with the db up) and the test shows how this would impact processing. At the very least this should be

```swift
            let targets: [(String, TargetType)]? = await {
                guard let repo = package.repository, let owner = repo.owner, let repository = repo.name else {
                    return nil
                }
                do {
                    return try? await API.PackageController.Target.query(on: database, owner: owner, repository: repository)
                }
            }()

```

The timeout is happening, because it's using `database` and we're inside a transaction - leading to a deadlock until it times out.

So it needs to be

```swift
            let targets: [(String, TargetType)]? = await {
                guard let repo = package.repository, let owner = repo.owner, let repository = repo.name else {
                    return nil
                }
                do {
                    return try? await API.PackageController.Target.query(on: tx, owner: owner, repository: repository)
                }
            }()
```

(I'd also avoid the immediately executing close here but I suspect that's just WIP.)

Reply:
thank you @daveverwer and @finestructure for debugging! 

@finestructure would you be able to clarify what you mean by the following? I would like to address it, but want to make sure I understand what is preferred.

> (I'd also avoid the immediately executing close here but I suspect that's just WIP.)

Reply:
thank you @finestructure for the detail writeup and helping to improve this PR! this definitely helps clarify your thought process with me and I definitely agree with the points made. I also prefer option 1, so I updated the PR to use a private static function instead. let me know if there's anything else that should be addressed :)